feat(kernels): lean Hub integration via Transformers passthrough (#3426)
NVIDIA-NeMo/Automodel: docs/guides/huggingface-api-compatibility.mdx

Reviewer: ```suggestion
- **API compatibility**: For many workflows, you can keep your existing `transformers` code and swap in NeMo AutoModel drop-in wrappers (`NeMoAutoModel*`, `NeMoAutoTokenizer`) with minimal changes.
- **Artifact compatibility**: NeMo AutoModel produces Hugging Face-compatible checkpoints (config, tokenizer, and safetensors) that can be loaded by Hugging Face Transformers and downstream tools such as vLLM and SGLang.
```

Reviewer: ```suggestion
Some downstream environments might still run Transformers v4, while NeMo AutoModel development and tests now target Transformers v5.
```

Reviewer: ```suggestion
NeMo AutoModel provides `NeMoAutoTokenizer` as a Transformers-like auto-tokenizer with a small registry for specialized backends and a safe fallback when no specialization is needed.
```

Reviewer: ```suggestion
If your goal is to train or fine-tune in NeMo AutoModel and deploy in the Hugging Face ecosystem, the recommended workflow is to keep `model_save_format: safetensors` with either `save_consolidated: final` for final-checkpoint export or `save_consolidated: false` plus `bash <checkpoint>/model/consolidate.sh` after training. Then, load `model/consolidated/` using standard Hugging Face APIs or downstream inference engines. Set `save_consolidated: every` (or legacy `true`) only if you want inline Hugging Face export at every checkpoint save.
```

NeMo AutoModel introduces several optimizations and architectural changes to improve training efficiency and hardware utilization.

- **Performance features**: NeMo AutoModel can automatically apply optional kernel patches and optimizations (for example SDPA selection, Liger kernels, DeepEP) while keeping the public model API the same. Hugging Face Hub kernels are also supported via `attn_implementation: kernels-community/flash-attn2` and `use_kernels: true` — see [Hub Kernels](/development/hub-kernels).

> - **Distributed training stack**: NeMo AutoModel recipes and CLI are designed for multi-GPU and multi-node fine-tuning with PyTorch-native distributed features (FSDP2, pipeline parallelism, etc.). When loading models directly from Python, pass a [`DistributedSetup`](/development/distributed-setup) via `from_pretrained(..., distributed_setup=...)` to enable tensor, pipeline, context, and expert parallelism.

Reviewer: ```suggestion
- **Performance features**: NeMo AutoModel can automatically apply optional kernel patches and optimizations (for example, SDPA selection, Liger kernels, and DeepEP) while keeping the public model API the same. Hugging Face Hub kernels are also supported through `attn_implementation: kernels-community/flash-attn2` and `use_kernels: true`. See [Hub Kernels](/development/hub-kernels).
- **Distributed training stack**: NeMo AutoModel recipes and CLI are designed for multi-GPU and multi-node fine-tuning with PyTorch-native distributed features such as FSDP2 and pipeline parallelism. When loading models directly from Python, pass a [`DistributedSetup`](/development/distributed-setup) through `from_pretrained(..., distributed_setup=...)` to enable tensor, pipeline, context, and expert parallelism.
```
